gcp_variant_transforms/beam_io/vcf_header_io.py

```diff
@@ -193,3 +194,86 @@ def __init__(
 
   def expand(self, pvalue):
     return pvalue.pipeline | Read(self._source)
+
+
+class _WriteVcfHeaderFn(beam.DoFn):
+  """A DoFn for writing VCF headers to a file."""
+
+  HEADER_TEMPLATE = '##{}=<{}>\n'
+  FINAL_HEADER_LINE = '#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT\n'
+  INFO_KEY = 'INFO'
+  FILTER_KEY = 'FILTER'
+  ALT_KEY = 'ALT'
+  FORMAT_KEY = 'FORMAT'
+  CONTIG_KEY = 'contig'
+  ID_KEY = 'ID'
+  NUMBER_KEY = 'Number'
+  TYPE_KEY = 'Type'
+  DESCRIPTION_KEY = 'Description'
+
+  def __init__(self, file_path):
+    self._file_path = file_path
+
+  def process(self, header):
+    with FileSystems.create(self._file_path) as file_to_write:
+      self._write_headers_by_key(file_to_write, self.INFO_KEY, header.infos)
+      self._write_headers_by_key(file_to_write, self.FILTER_KEY, header.filters)
+      self._write_headers_by_key(file_to_write, self.ALT_KEY, header.alts)
+      self._write_headers_by_key(file_to_write, self.FORMAT_KEY, header.formats)
+      self._write_headers_by_key(file_to_write, self.CONTIG_KEY, header.contigs)
+      file_to_write.write(self.FINAL_HEADER_LINE)
+
+  def _write_headers_by_key(self, file_to_write, header_key, header):
+    if header_key == self.INFO_KEY:
+      self._remove_source_and_version(header)
+
+    for header_value in header.values():
+      file_to_write.write(self._to_vcf_header_line(header_key, header_value))
+
+  def _to_vcf_header_line(self, header_key, header):
+    headers = ['{}={}'.format(*self._map_header(header, key)) for key in header]
+    return self.HEADER_TEMPLATE.format(header_key, ','.join(headers))
+
+  def _map_header(self, header, key):
+    value = header[key]
+    key = self._map_key(key)
+    if key == self.NUMBER_KEY:
+      value = self._map_num(value)
+    elif key == self.DESCRIPTION_KEY:
+      value = '"{}"'.format(value)
+    return (key, value)
+
+  def _map_key(self, key):
+    if key == 'id':
+      return self.ID_KEY
+    elif key == 'num':
+      return self.NUMBER_KEY
+    elif key == 'desc':
+      return self.DESCRIPTION_KEY
+    elif key == 'type':
+      return self.TYPE_KEY
+    else:
+      raise ValueError('Unknown VCF header key {}.'.format(key))
+
+  def _map_num(self, num):
```

Comment:
There are also 'G', 'R' types :)
Looks like we need to use this method here: https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/beam_io/vcfio.py#L795 ... I'm ok to copy it here since we'll be merging that function with Beam eventually.
But I think it's better to do this conversion inside _VcfHeaderSource? This way we eliminate all dependencies on PyVCF in one transform. Not sure how easy it is to do it there though..

Reply:
Sounds good. Do you have any sample header lines with 'G' or 'R'? I think it would probably be good to add all variations into the get_sample_vcf_header_lines function so that we consistently test this throughout our code.

Reply:
We have them as special cases here: https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/beam_io/vcfio_test.py#L389
Also, instead of hardcoding the values (-1, -2, etc) could you use the same logic as '_get_field_count_as_string' in vcfio? I worry about hardcoding pyvcf's values in our code as they could change in the future and new ones could be added and we would not know about it.

Reply:
Sounds good! Just made the switch.